celesta-core/src/main/java/ru/curs/celesta/event/TriggerDispatcher.java

```diff
@@ -8,25 +8,27 @@
 /**
  * Created by ioann on 31.05.2017.
  */
-public class TriggerDispatcher {
+public final class TriggerDispatcher {
 
-    private final EnumMap<TriggerType, Map<Class<? extends Cursor>, List<Consumer<Cursor>>>> triggerMap =
+    private final EnumMap<TriggerType, Map<Class<? extends Cursor>, List<Consumer<?>>>> triggerMap =
             new EnumMap<>(TriggerType.class);
 
     public TriggerDispatcher() {
         Arrays.stream(TriggerType.values()).forEach(t -> triggerMap.put(t, new HashMap<>()));
     }
 
-    public void registerTrigger(TriggerType type, Class<? extends Cursor> cursorClass, Consumer consumer) {
-        Map<Class<? extends Cursor>, List<Consumer<Cursor>>> cursorClassMap = triggerMap.get(type);
+    public <T extends Cursor> void registerTrigger(TriggerType type, Class<T> cursorClass, Consumer<? super T> consumer) {
```

Comment:
It'd be nice having some use case in form of a test where `? super T` would play a role.